flink-table/flink-table-common/src/main/java/org/apache/flink/table/catalog/CatalogDatabase.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.catalog;
+
+import java.util.Map;
+
+/**
+ * Represents a database object in a catalog.
+ */
+public interface CatalogDatabase {
+	/**
+	 * Get a map of properties associated with the database.
+	 */
+	Map<String, String> getProperties();
```

Comment:
Add more very common methods, I would suggest to at at least `getDescription(): Optional<String>` (usually one sentence for the CLI listing) and `getDetailedDescription(): Optional<String>` (a lot of text).

Reply:
Hi Timo, Thanks for the feedback.

By "description", do you mean "comment" that is given for a database or table? It's fine to add getComment() here. However, typical databases only allow user to give just one common. Therefore, I'm not sure if we need getDetailedDescription() and how one might implement it. Any further thoughts?

Reply:
I chose the naming `getDescription()` and `getDetailedDescription()` to be in sync with UDFs that will also receive this interfaces. I would also be fine with just `getDescription()`. However, I think it would be helpful to have a read-only method that returns a catalog dependent explanation string. For Hive, such a string could contain LOCATION, USER, ROLES etc. Information that might be interesting for users during a CLI session that is not part of the writable comment field.

Reply:
Okay, sounds good.